src/gordon_gcp/plugins/service/event_consumer.py

```diff
@@ -304,6 +315,21 @@ def _get_and_validate_pubsub_msg_schema(self, pubsub_msg_data):
             self._logger, {'msg_id': msg_id})
         msg_logger.info('Received new message.')
 
+        # The timestamp is in a weird format; see
+        # https://pythonhosted.org/gax-google-logging-v2/google.protobuf.timestamp_pb2.html
+        # It divides a time with nanosecond precision into an integer part and
+        # a nanoseconds part.  To turn them into a regular float, we need to
+        # divide the nanos by a billion (nano = billionth) and add.
+        # E.g.: 1234 seconds and 123456789 nanos -> 1234.123456789
+        msg_timestamp = (pubsub_msg.publish_time.seconds +
+                         (pubsub_msg.publish_time.nanos / 10**9))
+        msg_age = time.time() - msg_timestamp
+        if msg_age > self._max_msg_age:
+            msg_logger.warn(f'Message is too old ({msg_age} seconds), '
+                            'discarding and acking.')
+            pubsub_msg.ack()
+            return
```

Comment:
Could we maybe emit a metric here too?

Reply:
ok

Reply:
Uh am I losing it, or do we not actually have any metrics in this repo?  The metrics plugin is being passed everywhere but I don't see it being called.

Reply:
I would say I'll add a todo and we should have a ticket that covers all of these.

Reply:
Turns out adding the metric is super simple, so I'll just do it.